logbook-api/src/main/java/org/zalando/logbook/attributes/HttpAttributes.java

```diff
@@ -0,0 +1,77 @@
+package org.zalando.logbook.attributes;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.experimental.Delegate;
+import org.apiguardian.api.API;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.apiguardian.api.API.Status.STABLE;
+
+@API(status = STABLE)
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public final class HttpAttributes implements Map<String, String> {
+
+    public static final HttpAttributes EMPTY = new HttpAttributes(Collections.emptyMap());
+
+    @Delegate
+    private final Map<String, String> map;
+
+    public HttpAttributes() {
+        map = new ConcurrentHashMap<>();
+    }
+
+    public static HttpAttributes withMap(Map<String, String> map) {
+        return new HttpAttributes(new ConcurrentHashMap<>(map));
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof HttpAttributes)) return false;
+        return map.equals(o);
+    }
+
+    @Override
+    public int hashCode() {
+        return map.hashCode();
+    }
+
+    @Override
+    public String toString() {
+        return map.toString();
+    }
+
+    public HttpAttributes fluentPut(String key, String value) {
```

Comment:
Are those used anywhere? I'd ditch them in favor of making this class immutable.

Reply:
Removed all mutating methods; the class is now Immutable.